### PR TITLE
:sparkles: adding override provider settings flag

### LINKED ISSUE #220: Ability to use custom providers
We should have an option for users to pass in their own provider config, and use that to run analysis with the custom provider.

### PR DESCRIPTION
**Pull request by shawn-hurley**
Closes #220 
basically if you use the host DNS name in a provider settings like:

```
[
    {
        "name": "aws",
        "address": "host.containers.internal:10000",
        "initConfig": [
            {
                "location": "shurley-test",
            },
        ],
    },
]
```

notes:

This allows for non-input analysis, which makes sense because location and things must be described in the provider settings. This means we have to turn off the required flag.

### CODE REVIEW

File: cmd/analyze.go [hunk 1]
```diff
@@ -213,6 +222,7 @@ func NewAnalyzeCmd(log logr.Logger) *cobra.Command {
 	analyzeCommand.Flags().StringVar(&analyzeCmd.noProxy, "no-proxy", loadEnvInsensitive("no_proxy"), "proxy excluded URLs (relevant only with proxy)")
 	analyzeCommand.Flags().IntVar(&analyzeCmd.contextLines, "context-lines", 100, "number of lines of source code to include in the output for each incident")
 	analyzeCommand.Flags().StringVar(&analyzeCmd.incidentSelector, "incident-selector", "", "an expression to select incidents based on custom variables. ex: (!package=io.konveyor.demo.config-utils)")
+	analyzeCommand.Flags().StringVar(&analyzeCmd.overrideProviderSettings, "override-provider-settings", "", "override the provider settings, the analsis pod will be run on the host network and no providers with be started up")
```
Comment (eemcmullan) [inline cmd/analyze.go:225]: NIT: typo "analysis"

File: cmd/analyze.go [hunk 2]
```diff
@@ -832,6 +856,104 @@ func (a *analyzeCommand) RunProviders(ctx context.Context, networkName string, v
...
+	}
+
+	args := []string{
+		fmt.Sprintf("--provider-settings=%s", ProviderSettingsMountPath),
+		fmt.Sprintf("--output-file=%s", AnalysisOutputMountPath),
+		fmt.Sprintf("--context-lines=%d", a.contextLines),
+	}
+
+	a.enableDefaultRulesets = false
+	if a.enableDefaultRulesets {
```
Comment (eemcmullan) [inline cmd/analyze.go:883]: Would we also want to check which provider is being used? As right now, the default rulesets are only Java rules.

File: cmd/analyze.go [hunk 3]
```diff
@@ -832,6 +856,104 @@ func (a *analyzeCommand) RunProviders(ctx context.Context, networkName string, v
...
+		args = append(args,
+			fmt.Sprintf("--rules=%s/", CustomRulePath))
+	}
+
+	if a.jaegerEndpoint != "" {
+		args = append(args, "--enable-jaeger")
+		args = append(args, "--jaeger-endpoint")
+		args = append(args, a.jaegerEndpoint)
+	}
+	if !a.analyzeKnownLibraries {
```
Comment (eemcmullan) [inline cmd/analyze.go:903]: This fails analysis if provider does not have dependency cap. Unless we're assuming if you're overriding a provider, you also are aware of the options it supports

File: cmd/analyze.go [hunk 4]
```diff
@@ -832,6 +856,104 @@ func (a *analyzeCommand) RunProviders(ctx context.Context, networkName string, v
...
+			fmt.Sprintf("--dep-label-selector=(!%s=open-source)", provider.DepSourceLabel))
+	}
+	if a.logLevel != nil {
+		args = append(args, fmt.Sprintf("--verbose=%d", *a.logLevel))
+	}
+	labelSelector := a.getLabelSelector()
+	if labelSelector != "" {
+		args = append(args, fmt.Sprintf("--label-selector=%s", labelSelector))
+	}
+	if a.mode == string(provider.FullAnalysisMode) {
```
Comment (eemcmullan) [inline cmd/analyze.go:914]: same as above here

File: cmd/analyze.go [hunk 5]
```diff
@@ -124,61 +125,69 @@ func NewAnalyzeCmd(log logr.Logger) *cobra.Command {
...
+				}
+				err = analyzeCmd.RunAnalysis(cmd.Context(), xmlOutputDir, containerVolName, foundProviders, providerPorts)
+				if err != nil {
+					log.Error(err, "failed to run analysis")
+					return err
+				}
+			} else {
+				err := analyzeCmd.RunAnalysisOverrideProviderSettings(cmd.Context())
+				if err != nil {
+					log.Error(err, "failed to create json output file")
```
Comment (eemcmullan) [inline cmd/analyze.go:186]: NIT log message

File: cmd/analyze.go (hunk 3 above)
Comment (shawn-hurley) [reply cmd/analyze.go:903]: We have to make that assumption for now, as the Kantra command does not have the provider caps/specs.

File: cmd/analyze.go (hunk 2 above)
Comment (shawn-hurley) [reply cmd/analyze.go:883]: I think this is another case where the user has to know which options to set for their providers. 

I can see that we probably need to have some more docs, I wonder if this is necessary as we do your work. Can we do a follow-up that lays out a lot of these caveats when using this override setting?

File: cmd/analyze.go (hunk 4 above)
Comment (shawn-hurley) [reply cmd/analyze.go:1074]: Resolving based on comment above and agreement.

**Pull request merged**